packages/gotrue/lib/src/gotrue_client.dart

```diff
@@ -479,8 +479,52 @@ class GoTrueClient {
     return refreshCompleter.future;
   }
 
+  /// Sends a reauthentication OTP to the user's email or phone number.
+  ///
+  /// Requires the user to be signed-in.
+  Future<void> reauthenticate() async {
+    final session = currentSession;
+    if (session == null) {
+      throw AuthException('Not logged in.');
+    }
+
+    final options =
+        GotrueRequestOptions(headers: headers, jwt: session.accessToken);
+
+    await _fetch.request(
+      '$_url/reauthenticate',
+      RequestMethodType.get,
+      options: options,
+    );
+  }
+
+  ///Resends an existing signup confirmation email, email change email, SMS OTP or phone change OTP.
+  ///
+  ///Use [EmailResendParams] or [PhoneResendParams] to specify the type of resend.
+  Future<ResendResponse> resend(ResendParams params) async {
```

Comment:
I really like what you have going on here with the enum and two classes extending a base abstract class, but I personally think it would be easier for the users to use this method if the parameters were simple named parameters like this and add comment docs and assertion to make sure the user knows what value to pass or not pass. 

```suggestion
  Future<ResendResponse> resend({
    String? email,
    String? phone,
    OtpType type,
    String? captchaToken,
  }) async {
```

Reply:
Yeah I thought that as well and wasn't sure. Will change hat.